lms/djangoapps/courseware/tests/test_access.py

```diff
@@ -169,10 +202,22 @@ def test__has_access_descriptor_staff_lock(self):
         mock_unit.start = datetime.datetime.now(pytz.utc) + datetime.timedelta(days=1)  # release date in the future
         mock_unit.visible_to_staff_only = True
         self.verify_access(mock_unit, False)
+        self.verify_error_type(mock_unit, access_response.VisibilityError)
 
         # Start date in the future, staff lock off.
         mock_unit.visible_to_staff_only = False
         self.verify_access(mock_unit, False)
+        self.verify_error_type(mock_unit, access_response.StartDateError)
+
+    def test__has_access_descriptor_beta_user(self):
```

Comment:
Can you `ddt` this?  In fact, most of the tests in this file seem like they can be handled by a single ddt test function.  Possibly not milestones and preview tests, though.  I can sit with you on Monday if you need help refactoring the tests to do this.